tt_metal/impl/dispatch/dispatch_core_manager.hpp

```diff
@@ -388,7 +403,7 @@ class dispatch_core_manager {
     // Each device has an assigned hugepage at a specific channel that holds (up to 2) hardware command queues (represented by cq_id)
     std::unordered_map<chip_id_t, std::unordered_map<uint16_t, std::unordered_map<uint8_t, dispatch_core_placement_t>>> dispatch_core_assignments;
     std::unordered_map<chip_id_t, std::list<CoreCoord>> available_dispatch_cores_by_device;
-    std::unordered_map<chip_id_t, CoreType> dispatch_core_type_by_device;
+    std::unordered_map<chip_id_t, CoreType> dispatch_core_type_by_device;  //TODO: dispatch_core_type_by_device should probably be for all devices, not per device
```

Comment:
it seems `dispatch_core_manager` should eventually be moved to be at the DevicePool level then